### PR TITLE
feat: Add Owner during organization creation

### LINKED ISSUE #12998: [CAL-2855] 'Add Owner' step/section during creation of an organization.
As an owner is supposed to be able to see the billing details, it is important so that the person creating the organization(the cal.com instance admin) can make someone from the organization itself the owner.

As suggested by Sean, it could be an improvement to current invite admin flow during org creation and allow adding any member with any role.

Internal Thread:
[https://threads.com/thread/34559464671/34559464669?s=mpdF3D8unMY15AsXGz8fR1](https://threads.com/thread/34559464671/34559464669?s=mpdF3D8unMY15AsXGz8fR1)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2855](https://linear.app/calcom/issue/CAL-2855/add-owner-stepsection-during-creation-of-an-organization)</sub>

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

Fixes #12998

https://www.loom.com/share/02f3e2bc952b4f1b8316461b1ab94d98?sid=f62ac5d5-7b9b-4811-a03a-db153436e6d8

### CODE REVIEW

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 1]
```diff
@@ -170,6 +170,7 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
   const resetFields = () => {
     newMemberFormMethods.reset();
     newMemberFormMethods.setValue("emailOrUsername", "");
+    newMemberFormMethods.setValue("role", options[0].value);
```
Comment (SomayChauhan) [inline packages/features/ee/teams/components/MemberInvitationModal.tsx:173]: this is to fix a bug i encountered, notice the second time where the `invite as` default value is set to `member`, but the user ends up being invited as an `admin`. It seems like the previous value is persisting incorrectly.

https://www.loom.com/share/fa17f2ad127f41ff81b931a98963c7f2?sid=f06e1a9c-03c9-40c3-b090-3e53283efb4f

File: packages/features/ee/teams/components/AddNewTeamMembers.tsx [hunk 2]
```diff
@@ -144,14 +144,14 @@ export const AddNewTeamMembersForm = ({
                   onSuccess: async (data) => {
                     await utils.viewer.teams.get.invalidate();
                     setMemberInviteModal(false);
+                    resetFields();
```
Comment (SomayChauhan) [inline packages/features/ee/teams/components/AddNewTeamMembers.tsx:147]: probably should do this at other places too

**Review by Udit-takkar — COMMENTED** [review]
@SomayChauhan there is one E2E test that is failing "› Organization › Admin should be able to create an organization and complete onboarding ".

**Comment by SomayChauhan** [comment]
Hmm, 🤔🤔 I don't understand,
I changed the route from 'onboard-admins' to 'onboard-members' maybe that's why

File: apps/web/playwright/organization/organization-creation.e2e.ts [hunk 3]
```diff
@@ -103,26 +105,39 @@ test.describe("Organization", () => {
...
+      await expect(page.getByTestId("pending-member-item")).toHaveCount(2);
+
+      // can remove members
+      await expect(page.getByTestId("pending-member-item")).toHaveCount(2);
+      const lastRemoveMemberButton = page.getByTestId("remove-member-button").last();
+      await lastRemoveMemberButton.click();
+      await page.waitForLoadState("networkidle");
+      await expect(page.getByTestId("pending-member-item")).toHaveCount(1);
+      // Cleanup here since this user is created without our fixtures.
+      await prisma.user.delete({ where: { email: adminEmail } });
```
Comment (Udit-takkar) [inline apps/web/playwright/organization/organization-creation.e2e.ts:138]: Not sure if we should delete this user.
 
This was also used in organization-invitation.e2e.ts

      const adminEmail = users.trackEmail({ username: "rick", domain: `${orgDomain}.com` });
      
      but no cleanup  code there

**Review by Udit-takkar — COMMENTED** [review]
@SomayChauhan Great work. Left few comments. Everything else works fine

File: apps/web/playwright/organization/organization-creation.e2e.ts (hunk 3 above)
Comment (SomayChauhan) [reply apps/web/playwright/organization/organization-creation.e2e.ts:138]: > Not sure if we should delete this user.
> 
> This was also used in organization-invitation.e2e.ts
> 
> ```
>   const adminEmail = users.trackEmail({ username: "rick", domain: `${orgDomain}.com` });
>   
>   but no cleanup  code there
> ```

https://github.com/SomayChauhan/cal.com/blob/560288015c413f51fd7a2c7321b44beda4ce2443/apps/web/playwright/teams.e2e.ts#L340

copied it over from `Can create teams via Wizard` test

File: apps/web/playwright/organization/organization-creation.e2e.ts [hunk 4]
```diff
@@ -2,6 +2,8 @@ import { expect } from "@playwright/test";
 import path from "path";
 import { uuid } from "short-uuid";
 
+import { prisma } from "@calcom/prisma";
```
[the remark this answers is not in the document]
Comment (SomayChauhan) [reply apps/web/playwright/organization/organization-creation.e2e.ts:5]: also copied over from `teams.e2e.ts`

File: packages/features/ee/teams/components/AddNewTeamMembers.tsx [hunk 5]
```diff
@@ -278,7 +278,7 @@ const PendingMemberItem = (props: { member: TeamMember; index: number; teamId: n
           )}
         </div>
       </div>
-      {(member.role !== "OWNER" || isOrgAdminOrOwner) && (
+      {(member.role !== "OWNER" || isOrgAdminOrOwner) && member.id !== session.data?.user.id && (
```
Comment (SomayChauhan) [inline packages/features/ee/teams/components/AddNewTeamMembers.tsx:281]: > NIT: Remove the delete button icon for myself. I was able to delete org membership from /onboard-members

do we want this behaviour while creating new teams as well?

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**